airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/util/BigqueryClientFactory.kt

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.integrations.destination.bigquery.util
+
+import com.google.auth.oauth2.GoogleCredentials
+import com.google.cloud.bigquery.BigQuery
+import com.google.cloud.bigquery.BigQueryOptions
+import io.airbyte.integrations.destination.bigquery.BigQueryUtils
+import io.airbyte.integrations.destination.bigquery.spec.BigqueryConfiguration
+import io.github.oshai.kotlinlogging.KotlinLogging
+import io.micronaut.context.annotation.Factory
+import jakarta.inject.Singleton
+import java.io.ByteArrayInputStream
+import java.nio.charset.StandardCharsets
+
+private val logger = KotlinLogging.logger {}
+
+@Factory
+class BigqueryClientFactory(private val config: BigqueryConfiguration) {
```

Comment:
this class is just stolen from https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/destination-bigquery/src/main/kotlin/io/airbyte/integrations/destination/bigquery/BigQueryDestination.kt#L519-L558